test/distributed/test_multi_threaded_pg.py

```diff
@@ -249,7 +252,25 @@ def test_all_reduce_coalesced(self):
         self.assertEqual(t0, torch.ones(3, 3) * res_num)
         self.assertEqual(t1, torch.ones(3, 3) * (res_num * 2))
 
-# def all_reduce_coalesced(tensors, op=ReduceOp.SUM, group=None, async_op=False):
+    @skip_if_lt_x_gpu(1)
+    def test_bwd_sees_fwd_pg(self):
+
+        class MyFunc(torch.autograd.Function):
```

Comment:
can we have a test where we explicitly retrieve the process groups used in fwd and bwd and validate they are equivalent?

Reply:
actually, what is "forward_pg"? The test name implies there is a pg used in forward, but the forwrad function doesn't do any distributed compute so is `forward_pg` the same as `default pg`?

Reply:
The global PG of the fwd thread should match the one of the bwd thread.

Should be pretty easy to test for that, I'll change the test to do so once I figure out why it's crashing.